pkg/cmd/roachtest/prometheus.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"gopkg.in/yaml.v2"
+)
+
+// prometheusScrapeNode are nodes to scrape from.
+type prometheusScrapeNode struct {
+	node option.NodeListOption
+	port int
+}
+
+// prometheusConfig is a monitor that watches over the running of prometheus.
+type prometheusConfig struct {
+	prometheusNode option.NodeListOption
+	jobName        string
+	scrapeNodes    []prometheusScrapeNode
+}
+
+// run installs and runs prometheus.
+// Prometheus will run in the background and log an error if it fails, as opposed
+// to failing the test.
+func (pm *prometheusConfig) run(ctx context.Context, t *test, c Cluster) {
+	cfg, err := makePrometheusConfig(
+		ctx,
+		c,
+		pm.jobName,
+		pm.scrapeNodes,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := repeatRunE(
+		ctx,
+		t,
+		c,
+		pm.prometheusNode,
+		"install prometheus",
+		fmt.Sprintf(
+			`rm -rf /tmp/prometheus && mkdir /tmp/prometheus && cd /tmp/prometheus &&
+			wget -O prometheus.tar.gz https://github.com/prometheus/prometheus/releases/download/v2.27.1/prometheus-2.27.1.linux-amd64.tar.gz &&
```

Comment:
[nit] It doesn't matter, but we use `curl` nearly everywhere else for downloading things.  If you do move to curl, you'll want the `--fail` flag most likely.

Reply:
This currently won't work for local runs for non-linux workstations.  Are there differences other than the binary that we are using?  If so, perhaps we could add a TODO comment with some of the differences and output a better error message if you are trying to run on a local cluster and your `runtime.GOOS` is not linux.  If not, perhaps a small function that returns the right download URL for the given platform so that this would work on local clusters?